src/service/folders.rs

```diff
@@ -132,16 +132,15 @@ pub async fn list_folders(
 }
 
 #[tracing::instrument()]
-pub async fn get_folder(org_id: &str, folder_id: &str) -> Result<HttpResponse, Error> {
-    let resp = if let Ok(folder) = db::folders::get(org_id, folder_id).await {
-        HttpResponse::Ok().json(folder)
-    } else {
-        return Ok(HttpResponse::NotFound().json(MetaHttpResponse::error(
```

Comment:
The change here was necessary because I changed the return type of `db::folders::get` from `Result<Folder, _>` to `Result<Option<Folder>, _>`.

Related to that change, I think the previous code made the assumption that any returned `Err` value indicated that the folder does not exist. However, in theory the call to `db::folders::get` could fail for other reasons as well. That's why I split this up into a match with different cases